cylc/flow/task_pool.py

```diff
@@ -565,9 +565,31 @@ def spawn_successor_if_parentless(
             next_task = (
                 self._get_hidden_task_by_id(taskid)
                 or self._get_main_task_by_id(taskid)
-                or self.spawn_task(
-                    itask.tdef.name, next_point, itask.flow_nums)
             )
+            if next_task is None:
+                # Does not exist yet, spawn it.
+                next_task = self.spawn_task(
+                    itask.tdef.name, next_point, itask.flow_nums)
+            else:
+                # Already exists
+                retroactive_spawn = (
+                    not next_task.flow_nums
+                    and not next_task.state.is_runahead
+                )
+                next_task.merge_flows(itask.flow_nums)
+                LOG.info(
+                    f"[{next_task}] merged in flow(s) "
+                    f"{','.join(str(f) for f in itask.flow_nums)}"
+                )
+                if retroactive_spawn:
+                    # Did not belong to a flow (force-triggered) before merge.
+                    # Now it does, so spawn successor, and children if needed.
+                    LOG.info(
+                        f"[{next_task}] retroactive spawning post flow merge."
+                    )
```

Comment:
Not sure how useful this log line is, it seems to suggest `next_task` itself is being spawned, but it was already spawned before this

Reply:
I don't think users are likely to understand what this message means.

Reply:
`next_task` is spawning its children.

I think that is consistent with other log messages (need to check).  

I guess it is pretty cryptic!  Should be reworded and/or shifted to debug level. But the point is, there should be some explanation of why the scheduler suddenly spawns on task outputs that were completed some time ago, which is unusual behaviour.

Reply:
I think the "merged in flows" line covers that base.

Suggest demoting this to debug.

Reply:
demoted